app/serializers/search_and_compare/course_serializer.rb

```diff
@@ -82,8 +82,107 @@ class CourseSerializer < ActiveModel::Serializer
     attribute(:ProviderLocation)                      { provider_location }
     attribute(:ContactDetails)                        { contact_details }
 
+    # DescriptionSections_Mapping
+    attribute(:DescriptionSections)                   { description_sections }
+
   private
 
+    def default_description_section_value
+      {
+        Id: 0,
+        Ordinal: 0,
+        CourseId: 0,
+        Course: nil,
+      }
+    end
+
+    def text(enrichment_type, enrichment_key)
+      if enrichment_type == :course_enrichment
+        course_enrichment.send enrichment_key
+      elsif enrichment_type == :provider_enrichment
+        provider_enrichment.send enrichment_key
+      else
+        object.send enrichment_key
+      end
+    end
+
+    def description_sections
+      description_sections_enrichment_mappings.map do |mappings|
+        {
+          **default_description_section_value,
+          Name: mappings[:name],
+          Text: text(mappings[:enrichment_type], mappings[:enrichment_key])
+        }
```

Comment:
As discussed IRL, let's inline this into `description_sections_enrichment_mappings`